### PR TITLE
fix for unknown measure

### LINKED ISSUE #170: A custom/unknonw measure fails when read from server
## Describe the bug
When custom/unknown measure is added a later request of measure is requested from Atum server if fails with exception.

## To Reproduce
Steps to reproduce the behavior OR commands run:
1. Add a `UnknownMeasure` to an `AtumPartitioing`
2. Request the `AtumContext`
3. Exception is thrown

## Responsible code
```scala
  private def createMeasure(measure: MeasureDTO): za.co.absa.atum.agent.model.AtumMeasure = {
    val measuredColumns = measure.measuredColumns

    measure.measureName match {
      case RecordCount.measureName            => RecordCount()
      case DistinctRecordCount.measureName    => DistinctRecordCount(measuredColumns)
      case SumOfValuesOfColumn.measureName    => SumOfValuesOfColumn(measuredColumns.head)
      case AbsSumOfValuesOfColumn.measureName => AbsSumOfValuesOfColumn(measuredColumns.head)
      case SumOfHashesOfColumn.measureName    => SumOfHashesOfColumn(measuredColumns.head)
      case unsupportedMeasure =>
        throw MeasureException(
          s"Measure not supported: $unsupportedMeasure. Supported measures are: ${AtumMeasure.supportedMeasureNames}"
        )
    }
  }
```

## Expected behavior
Instance of `UnknownMeasure` is created. And it won't be evaluated on `createCheckpoint` call.

### PR DESCRIPTION
**Pull request by salamonpavel**
Fixes #170

### CODE REVIEW

File: agent/src/test/scala/za/co/absa/atum/agent/model/MeasuresBuilderTest.scala [hunk 1]
```diff
@@ -50,7 +49,7 @@ class MeasuresBuilderTest extends AnyFlatSpecLike {
       MeasureDTO("unsupportedMeasure", Seq("col"))
     )
 
-    assertThrows[MeasureException](MeasuresBuilder.mapToMeasures(unsupportedMeasure))
+    assert(MeasuresBuilder.mapToMeasures(unsupportedMeasure).isEmpty)
```
Comment (lsulak) [inline agent/src/test/scala/za/co/absa/atum/agent/model/MeasuresBuilderTest.scala:52]: see line 47, the test description needs to ba changed as well

File: agent/src/main/scala/za/co/absa/atum/agent/model/MeasuresBuilder.scala [hunk 2]
```diff
@@ -16,34 +16,42 @@
...
+private [agent] object MeasuresBuilder extends Logging {
 
   private [agent] def mapToMeasures(measures: Set[MeasureDTO]): Set[za.co.absa.atum.agent.model.AtumMeasure] = {
-    measures.map(createMeasure)
+    measures.flatMap { measure =>
+      createMeasure(measure) match {
+        case Some(value) =>
+          Some(value)
+        case None =>
+          logInfo(s"Measure not supported or unknown: $measure.")
```
Comment (lsulak) [inline agent/src/main/scala/za/co/absa/atum/agent/model/MeasuresBuilder.scala:37]: I'm thinking whether we want info or warning here? Might be beneficial to have it a bit more visible in logs perhaps?

Comment (lsulak) [reply agent/src/main/scala/za/co/absa/atum/agent/model/MeasuresBuilder.scala:37]: What do you think?

**Review by benedeki — DISMISSED** [review]
- [x] code reviewed
- [ ] pulled
- [ ] built
- [ ] run

**Review by TebaleloS — DISMISSED** [review]

File: agent/src/test/scala/za/co/absa/atum/agent/model/MeasuresBuilderTest.scala (hunk 1 above)
Comment (salamonpavel) [reply agent/src/test/scala/za/co/absa/atum/agent/model/MeasuresBuilderTest.scala:52]: ok, reformulated

File: agent/src/main/scala/za/co/absa/atum/agent/model/MeasuresBuilder.scala (hunk 2 above)
Comment (salamonpavel) [reply agent/src/main/scala/za/co/absa/atum/agent/model/MeasuresBuilder.scala:37]: I am fine with both really, changing to warn. then.

**Review by lsulak — APPROVED** [review]

**Pull request merged**